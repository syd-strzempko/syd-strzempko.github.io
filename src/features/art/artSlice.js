import { createSlice } from '@reduxjs/toolkit';
import { importArt } from '../../utility/utils';

export const slice = createSlice({
    name: 'art',
    initialState: importArt(require.context('../../assets/art/', true, /\.jpg$/)),
    reducers: {},
});
  
// List selector

export const selectArts = (state) => {
    return state.art;
};

// Element selector

export const selectArt = (state, id) => {
    return state.art.find((single) => single.id.toString() === id);
};

export default slice.reducer;
  