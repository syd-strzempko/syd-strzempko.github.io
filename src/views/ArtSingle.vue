 <template>
  <div class="content">
    <router-link to="/art">Back to List</router-link>
    <div v-if="data" class="carousel">
      <router-link v-if="!isLast" :to="{ name: 'artsingle', params: { id: (data.art.id + 1) }}">
        <span class="iconlink">↩</span>
      </router-link>
      <span v-else class="iconlink empty">↩</span>
      <img :src="data.art.pathLong" />
      <router-link v-if="!isFirst" :to="{ name: 'artsingle', params: { id: (data.art.id - 1) }}">
        <span class="iconlink">↪</span>
      </router-link>
      <span v-else class="iconlink empty">↪</span>
    </div>
  </div>
</template>

<script>
export default {
  name: 'ArtSingle',
  props: {
    data: Object
  },
  computed: {
    isFirst() {
      return this.data.art.id == 0;
    },
    isLast() {
      return this.data.art.id == (this.data.total - 1);
    }
  },
}
</script>
<style scoped>
.content {
  display: flex;
  height: 100%;
  flex-flow: column nowrap;
  justify-content: center;
  padding: .25rem 0 .75rem;
}
.carousel {
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
}
img {
  flex: 1;
  min-width: 0;
}
a {
  height: 100%;
  display: flex;
  flex-flow: column;
  justify-content: center;
}
.empty {
  pointer-events: none;
  visibility: hidden;
}
</style>
