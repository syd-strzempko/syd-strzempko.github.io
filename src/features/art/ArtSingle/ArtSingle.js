import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import Layout from '../../../components/Layout';
import { selectArt } from '../artSlice';
import styles from '../art.module.css';
import ProgressiveImage from '../../../components/ProgressiveImage';

export const ArtSingle = () => {
  let { id } = useParams();
  const art = useSelector((state) => selectArt(state, id));
  return (
    <Layout header={'Art Project'} returnURL={'/art'}>
      <div className={styles.wrapperSingle}>
        <ProgressiveImage image={art} />
      </div>
    </Layout>
  );
}