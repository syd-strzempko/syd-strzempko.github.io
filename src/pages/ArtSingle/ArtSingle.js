import { Fragment } from 'react';
import Layout from '../../components/Layout';
import NavBar from '../../components/NavBar';
import './ArtSingle.css';

const ArtSingle = () => {
  return (
    <Layout>
      <NavBar />
      <Fragment>ArtSingle</Fragment>
    </Layout>
  );
}

export default ArtSingle;