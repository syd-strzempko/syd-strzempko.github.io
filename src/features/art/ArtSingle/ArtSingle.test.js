import { ArtSingle } from './ArtSingle';
import { MemoryRouter, useParams } from 'react-router-dom';
import renderer from 'react-test-renderer';
import * as reactRedux from 'react-redux';
import { mockStore } from '../artSlice.test';

jest.mock('../../../components/Circles/Circles');

// Note: MemoryRouter allows for nested Link components to exist inside of & access stack of mocked router
describe('ArtSingle', () => {
    // Mock useSelectors
    beforeEach(() => { useSelectorMock.mockImplementation(selector => selector(mockStore)); });
    afterEach(() => { useSelectorMock.mockClear(); });

    const useSelectorMock = reactRedux.useSelector;

    // Unit tests
    test('renders correctly', () => {
        useParams.mockReturnValue({ id: '0' });
        let view = renderer.create(
            <MemoryRouter>
                <ArtSingle />
            </MemoryRouter>
        ).toJSON();
        expect(view).toMatchSnapshot();
    });
});