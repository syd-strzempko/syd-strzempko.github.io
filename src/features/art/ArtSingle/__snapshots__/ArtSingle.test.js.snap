// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtSingle renders correctly 1`] = `
Array [
  <mock-Circles />,
  <div
    className="layout"
  >
    <div
      className="navWrapper"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/work"
        onClick={[Function]}
      >
        Work
      </a>
      <a
        href="/art"
        onClick={[Function]}
      >
        Art
      </a>
    </div>
    <div
      className="navClose"
      onClick={[Function]}
    >
      X
    </div>
    <div
      className="navOpen visible"
      onClick={[Function]}
    >
      <img
        alt="menu"
        src="hamburger.png"
      />
    </div>
    <a
      className="backLink"
      href="/art"
      onClick={[Function]}
    >
      <img
        alt="back"
        src="arrow.svg"
      />
    </a>
    <div
      className="wrapperSingle"
    >
      <img
        alt="work 0"
        className="image"
        src="mock_long.jpg"
        style={
          Object {
            "maxHeight": "calc(768px - 50px)",
          }
        }
      />
    </div>
  </div>,
]
`;
